AI Image Generation is Busted

### Parent Issue

_No response_

### Problem Statement
```
com.dotmarketing.exception.DotRuntimeException: Error generating image:com.dotcms.ai.exception.DotAIClientConnectException: Error while sending request to OpenAI
	at com.dotcms.ai.api.OpenAIImageAPIImpl.sendRequest(OpenAIImageAPIImpl.java:78) ~[?:?]
	at com.dotcms.ai.api.OpenAIImageAPIImpl.sendRequest(OpenAIImageAPIImpl.java:93) ~[?:?]
	at com.dotcms.ai.rest.ImageResource.handleImageRequest(ImageResource.java:120) ~[?:?]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52) ~[?:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124) ~[?:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167) ~[?:?]
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176) ~[?:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79) ~[?:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:469) ~[?:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:391) ~[?:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:80) ~[?:?]
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:253) ~[?:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248) ~[?:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244) ~[?:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292) ~[?:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274) ~[?:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244) ~[?:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265) ~[?:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:232) ~[?:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:679) ~[?:?]
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:392) ~[?:?]
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:346) ~[?:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:365) ~[?:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:318) ~[?:?]
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at com.dotmarketing.filters.CMSFilter.doFilterInternal(CMSFilter.java:205) ~[?:?]
	at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:59) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at com.dotcms.visitor.filter.servlet.VisitorFilter.doFilter(VisitorFilter.java:81) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at com.dotcms.vanityurl.filters.VanityURLFilter.doFilter(VanityURLFilter.java:107) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:176) ~[?:?]
	at org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:145) ~[?:?]
	at org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:92) ~[?:?]
	at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:389) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at com.dotmarketing.filters.TimeMachineFilter.doFilter(TimeMachineFilter.java:149) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at com.dotmarketing.filters.ThreadNameFilter.doFilter(ThreadNameFilter.java:88) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at com.dotmarketing.filters.CookiesFilter.doFilter(CookiesFilter.java:53) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at com.dotmarketing.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:99) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178) ~[?:?]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153) ~[?:?]
	at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[?:?]
Caused by: com.dotcms.ai.exception.DotAIClientConnectException: Error while sending request to OpenAI
	at com.dotcms.ai.client.openai.OpenAIClient.sendRequest(OpenAIClient.java:173) ~[?:?]
	at com.dotcms.ai.client.AIModelFallbackStrategy.doSend(AIModelFallbackStrategy.java:114) ~[?:?]
	at com.dotcms.ai.client.AIModelFallbackStrategy.lambda$sendRequest$8db58fd7$1(AIModelFallbackStrategy.java:181) ~[?:?]
	at io.vavr.control.Try.of(Try.java:75) ~[?:?]
	at com.dotcms.ai.client.AIModelFallbackStrategy.sendRequest(AIModelFallbackStrategy.java:181) ~[?:?]
	at com.dotcms.ai.client.AIModelFallbackStrategy.applyStrategy(AIModelFallbackStrategy.java:65) ~[?:?]
	at com.dotcms.ai.client.AIProxiedClient.sendToAI(AIProxiedClient.java:76) ~[?:?]
	at com.dotcms.ai.client.AIProxyClient.lambda$callToAI$0(AIProxyClient.java:73) ~[?:?]
	at java.base/java.util.Optional.map(Optional.java:260) ~[?:?]
	at com.dotcms.ai.client.AIProxyClient.callToAI(AIProxyClient.java:73) ~[?:?]
	at com.dotcms.ai.client.AIProxyClient.callToAI(AIProxyClient.java:98) ~[?:?]
	at com.dotcms.ai.client.AIProxyClient.callToAI(AIProxyClient.java:109) ~[?:?]
	at com.dotcms.ai.api.OpenAIImageAPIImpl.doRequest(OpenAIImageAPIImpl.java:178) ~[?:?]
	at com.dotcms.ai.api.OpenAIImageAPIImpl.sendRequest(OpenAIImageAPIImpl.java:63) ~[?:?]
	... 100 more
Caused by: net.jodah.failsafe.FailsafeException: java.net.SocketTimeoutException: Read timed out
	at net.jodah.failsafe.SyncFailsafe.call(SyncFailsafe.java:159) ~[?:?]
	at net.jodah.failsafe.SyncFailsafe.run(SyncFailsafe.java:93) ~[?:?]
	at com.dotcms.ai.client.AIProxyClient.callToAI(AIProxyClient.java:109) ~[?:?]
	at com.dotcms.ai.api.OpenAIImageAPIImpl.doRequest(OpenAIImageAPIImpl.java:178) ~[?:?]
	at com.dotcms.ai.api.OpenAIImageAPIImpl.sendRequest(OpenAIImageAPIImpl.java:63) ~[?:?]
	... 100 more
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:278) ~[?:?]
	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:304) ~[?:?]
	at com.dotcms.ai.api.OpenAIImageAPIImpl.sendRequest(OpenAIImageAPIImpl.java:63) ~[?:?]
	... 100 more
```

### Steps to Reproduce

Try to generate an ai image

### Acceptance Criteria

Make it work

### dotCMS Version

current

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

--- a/dotCMS/src/main/java/com/dotcms/ai/client/openai/OpenAIClient.java
+++ b/dotCMS/src/main/java/com/dotcms/ai/client/openai/OpenAIClient.java
@@ -49,10 +49,10 @@ import java.util.concurrent.atomic.AtomicLong;
  *
  * @auhor vico
  */
-public class OpenAIClient implements AIClient, EventSubscriber<SystemTableUpdatedKeyEvent> {
+public class OpenAIClient implements AIClient {
 
-    private static final String AI_OPEN_AI_TIMEOUT_KEY = "AI_PROVIDERS_OPENAI_TIMEOUT";
-    private static final String AI_OPEN_AI_ATTEMPTS_KEY = "AI_PROVIDERS_OPENAI_ATTEMPTS";
+    private static final String AI_OPEN_AI_TIMEOUT_KEY = "AI_OPEN_AI_TIMEOUT_KEY";
+    private static final String AI_OPEN_AI_ATTEMPTS_KEY = "AI_OPEN_AI_ATTEMPTS_KEY";
     private static final Lazy<OpenAIClient> INSTANCE = Lazy.of(OpenAIClient::new);
 
     public static OpenAIClient get() {
@@ -60,11 +60,11 @@ public class OpenAIClient implements AIClient, EventSubscriber<SystemTableUpdate
     }
 
     private static long resolveTimeout() {
-        return Config.getLongProperty(AI_OPEN_AI_TIMEOUT_KEY, 5000L);
+        return Config.getLongProperty(AI_OPEN_AI_TIMEOUT_KEY, 120 * 1000L);
     }
 
     private static int resolveAttempts() {
-        return Config.getIntProperty(AI_OPEN_AI_ATTEMPTS_KEY, 5);
+        return Config.getIntProperty(AI_OPEN_AI_ATTEMPTS_KEY, 3);
     }
 
     private static CircuitBreakerUrl.Method resolveMethod(final JSONObjectAIRequest request) {
@@ -83,23 +83,8 @@ public class OpenAIClient implements AIClient, EventSubscriber<SystemTableUpdate
         }
     }
 
-    private final AtomicLong openAiTimeout;
-    private final AtomicInteger openAiAttempts;
 
-    private OpenAIClient() {
-        openAiTimeout = new AtomicLong(resolveTimeout());
-        openAiAttempts = new AtomicInteger(resolveAttempts());
-    }
 
-    @Override
-    public void notify(final SystemTableUpdatedKeyEvent event) {
-        Logger.info(this, String.format("Notify for event [%s]", event.getKey()));
-        if (event.getKey().contains(AI_OPEN_AI_TIMEOUT_KEY)) {
-            openAiTimeout.set(resolveTimeout());
-        } else if (event.getKey().contains(AI_OPEN_AI_ATTEMPTS_KEY)) {
-            openAiAttempts.set(resolveAttempts());
-        }
-    }
 
     /**
      * {@inheritDoc}
@@ -153,8 +138,8 @@ public class OpenAIClient implements AIClient, EventSubscriber<SystemTableUpdate
                     .setAuthHeaders(AIModels.BEARER + appConfig.getApiKey())
                     .setUrl(jsonRequest.getUrl())
                     .setRawData(payload.toString())
-                    .setTimeout(openAiTimeout.get())
-                    .setTryAgainAttempts(openAiAttempts.get())
+                    .setTimeout(resolveTimeout())
+                    .setTryAgainAttempts(resolveAttempts())
                     .setOverrideException(statusCode -> resolveException(jsonRequest, modelName, statusCode))
                     .setRaiseFailsafe(true)
                     .build()

